pennylane/transforms/transpile.py

```diff
@@ -0,0 +1,156 @@
+"""
+Contains the transpiler transform.
+"""
+from typing import Union, List
+import networkx as nx
+
+from pennylane import apply
+from pennylane.ops.qubit import SWAP
+from pennylane.ops import __all__ as all_ops
+from pennylane.tape import QuantumTape
+from pennylane.transforms import qfunc_transform
+from pennylane.tape import stop_recording
+from pennylane.wires import Wires
+
+
+@qfunc_transform
+def transpile(tape: QuantumTape, coupling_map: Union[List, nx.Graph]):
+    """ Transpile a circuit according to a desired coupling map
+
+    Args:
+        tape (function): A quantum function.
+        coupling_map (list[tuple(int, int)] or nx.Graph): Either a list of tuples(int, int) or an instance of
+            networkx.Graph specifying the couplings between different qubits.
+    Returns:
+        function: the transformed quantum function
+
+    **Example**
+
+    Consider the following example circuit
+
+    .. code-block:: python
+
+        def circuit():
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[1, 3])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[0, 3])
+            return qml.probs(wires=[0, 1, 2, 3])
+
+    which, before transpiling it looks like this:
+
+        0: ──╭C──────────────╭C──╭┤ Probs
+        1: ──╰X──╭C──╭C──────│───├┤ Probs
+        2: ──╭C──│───╰X──╭C──│───├┤ Probs
+        3: ──╰X──╰X──────╰X──╰X──╰┤ Probs
```

Comment:
I can't verify as the function docstring isn't yet appearing in the documentation, but I have a feeling you will need to do this to have it render correctly :)
```suggestion
    which, before transpiling it looks like this:

    .. code-block:: text

        0: ──╭C──────────────╭C──╭┤ Probs
        1: ──╰X──╭C──╭C──────│───├┤ Probs
        2: ──╭C──│───╰X──╭C──│───├┤ Probs
        3: ──╰X──╰X──────╰X──╰X──╰┤ Probs
```

The same with the connectivity constraints diagram below.

Reply:
this does the trick:)